cmake/DaemonFlags.cmake

```diff
@@ -201,256 +184,350 @@ if (MSVC)
+	elseif (APPLE)
+		# Use libc++ on Mac because the shipped libstdc++
+		# version is too old
+		set_c_cxx_flag("-stdlib=libc++")
+		set_linker_flag("-stdlib=libc++")
+	endif()
+
+	# Prevent the generation of STB_GNU_UNIQUE symbols
+	# on templated functions on Linux.
+	# STB_GNU_UNIQUE renders dlclose() inoperative.
+	try_cxx_flag(FNO_GNU_UNIQUE "-fno-gnu-unique")
+
+	# Use MSVC-compatible bitfield layout
+	if (WIN32)
+		set_c_cxx_flag("-mms-bitfields")
+	endif()
+
+	# Linker flags
+	if (NOT APPLE)
+		try_linker_flag(LINKER_O1 "-Wl,-O1")
+		try_linker_flag(LINKER_SORT_COMMON "-Wl,--sort-common")
+		try_linker_flag(LINKER_AS_NEEDED "-Wl,--as-needed")
+
+		if (NOT USE_ADDRESS_SANITIZER)
+			try_linker_flag(LINKER_NO_UNDEFINED "-Wl,--no-undefined")
+		endif()
+
+		try_linker_flag(LINKER_Z_RELRO "-Wl,-z,relro")
+		try_linker_flag(LINKER_Z_NOW "-Wl,-z,now")
+	endif()
+
+	if (WIN32)
+		try_linker_flag(LINKER_DYNAMICBASE "-Wl,--dynamicbase")
+		try_linker_flag(LINKER_NXCOMPAT "-Wl,--nxcompat")
+		try_linker_flag(LINKER_LARGE_ADDRESS_AWARE "-Wl,--large-address-aware")
+		try_linker_flag(LINKER_HIGH_ENTROPY_VA "-Wl,--high-entropy-va")
+	endif()
+
+	if (MINGW AND USE_BREAKPAD)
+	    set_linker_flag("-Wl,--build-id")
+	endif()
+
+	# The -pthread flag sets some preprocessor defines,
+	# it is also used to link with libpthread on Linux
+	if (NOT APPLE)
+		try_c_cxx_flag(PTHREAD "-pthread")
+	endif()
+
+	if (USE_ADDRESS_SANITIZER)
+		set_cxx_flag("-fsanitize=address")
+		set_linker_flag("-fsanitize=address")
+	endif()
+
+	# Hardening.
+	if (USE_HARDENING OR NOT MINGW)
+		# MinGW with _FORTIFY_SOURCE and without -fstack-protector
+		# causes unsatisfied dependency on libssp.
+		# https://github.com/msys2/MINGW-packages/issues/5868
+		set_c_cxx_flag("-D_FORTIFY_SOURCE=2" RELEASE)
+		set_c_cxx_flag("-D_FORTIFY_SOURCE=2" RELWITHDEBINFO)
+		set_c_cxx_flag("-D_FORTIFY_SOURCE=2" MINSIZEREL)
+		# Don't set _FORTIFY_SOURCE in debug builds.
+	endif()
+
+	if (USE_HARDENING)
+		message(STATUS "Enabling hardening")
+
+		# PNaCl accepts the flags bug do not define __stack_chk_guard and __stack_chk_fail.
+		if (NOT NACL)
+			try_c_cxx_flag(FSTACK_PROTECTOR_STRONG "-fstack-protector-strong")
+
+			if (NOT FLAG_FSTACK_PROTECTOR_STRONG)
+				try_c_cxx_flag(FSTACK_PROTECTOR_ALL "-fstack-protector-all")
+			endif()
+		endif()
+
+		try_c_cxx_flag(FNO_STRICT_OVERFLOW "-fno-strict-overflow")
+		try_c_cxx_flag(WSTACK_PROTECTOR "-Wstack-protector")
+		try_c_cxx_flag(FPIE "-fPIE")
+		try_linker_flag(LINKER_PIE "-pie")
+
+		if ("${FLAG_LINKER_PIE}" AND MINGW)
+			# https://github.com/msys2/MINGW-packages/issues/4100
+			if (ARCH STREQUAL "i686")
+				set_linker_flag("-Wl,-e,_mainCRTStartup")
+			elseif(ARCH STREQUAL "amd64")
+				set_linker_flag("-Wl,-e,mainCRTStartup")
+			else()
+				message(FATAL_ERROR "Unsupported architecture ${ARCH}")
+			endif()
+		endif()
+	endif()
+
+	# Link-time optimizations.
+	if (USE_LTO)
+		message(STATUS "Enabling link time optimizations")
```

Comment:
The combination of adding these verbose messages, and passing unimplemented-in-NaCl build options such as USE_ARCH_INTRINSICS or USE_LTO to the NaCl build is unfortunate. It just causes the sub-build to lie with false messages such "Enabling link time optimizations"

Reply:
I removed the “Enabling link time optimizations” message, it was not saying the feature was enabled but the user asked for it so it was wrong anyway.
 
 Passing the `USE_LTO` option doesn't sound problematic. If the compiler lies about supporting LTO, it's not our fault. I also turned the `flto` setting as a “try”, so even setting `USE_LTO` on compilers that would reject the flag wouldn't break the build, and CMake will report the status. PNaCl will lie but that's not our problem.

The `USE_ARCH_INTRINSICS` would make sense the day we use Wasm, as Wasm support some kind of SIMD. Right now it's a bit useless for PNaCL but PNaCl not doing anything with it doesn't mean it's wrong to pass the option, in case the toolchain can do something with it.

Reply:
USE_LTO is not passed to NACL sub-build, also the option is ignored if for some reason the option is passed (like doing some manual stuff). After investigation, it happens that neither PNaCl underlying clang nor Saigo clang support `-flto`. It happens that the `pnacl-clang` wrapper accepts the option but I don't see how it would do something with it, and the produced binaries are the sames anyway.